File: fixtures/validation/scoring/ips/bad-repo/src/contracts/order-contract.ts
import { InternalOrder } from "../internal/order-entity";
import { logInfo } from "../infrastructure/logger";

export class OrderContract {
  constructor(
    public payload: any,
    public entity: InternalOrder
  ) {}
}

export function normalizeOrder(input: any): InternalOrder {
  logInfo(String(input?.id));
  return new InternalOrder(String(input?.id));
}

File: fixtures/dart-support/architecture-design/bad-repo/lib/contracts/order_contract.dart
import "../infrastructure/logger.dart";
import "../internal/order_entity.dart";

class OrderContract {
  final dynamic payload;
  final OrderEntity entity;

  OrderContract(this.payload, this.entity);
}

OrderEntity normalizeOrder(dynamic input) {
  logInfo("$input");
  return OrderEntity("$input");
}

